qa/tasks/cephfs/test_client_limits.py

```diff
@@ -82,19 +84,30 @@ def _test_client_pin(self, use_subdir):
 
         # The remaining caps should comply with the numbers sent from MDS in SESSION_RECALL message,
         # which depend on the caps outstanding, cache size and overall ratio
-        self.wait_until_equal(
-            lambda: self.get_session(mount_a_client_id)['num_caps'],
-            int(open_files * 0.2),
-            timeout=30,
-            reject_fn=lambda x: x < int(open_files*0.2))
+        def expected_caps():
+            num_caps = self.get_session(mount_a_client_id)['num_caps']
+            if num_caps < mds_min_caps_per_client:
+                raise
+            elif num_caps == mds_min_caps_per_client:
+                return True
+            elif num_caps == int((1.0-mds_max_ratio_caps_per_client)*(open_files+2)):
```

Comment:
I'm a little worry about the equal check. I think it's better to do range check.

Reply:
Ya makes sense, I'll fix it.

Reply:
https://github.com/ceph/ceph/pull/18041